GillianCore/debugging/utils/UnifyMap.ml

```diff
@@ -0,0 +1,188 @@
+module L = Logging
+module DL = Debugger_log
+open Syntaxes.Option
+
+type rid = L.ReportId.t [@@deriving show, yojson]
+type unify_result = Success | Failure [@@deriving yojson]
+
+module Make (Verification : Verifier.S) = struct
+  open Verification.SUnifier.Logging
+  open L.LoggingConstants
+
+  type result = unify_result = Success | Failure [@@deriving yojson]
```

Comment:
I think it was because I wanted the type both inside and outside the functor, and I couldn't use `nonrec` because it messes with yojson derivation.
I made a cleaner workaround in 2aed2bba9e2ba84204c192f059d99405207317f2.